### ISSUE TITLE
Home Assistant Addon

### ISSUE THREAD

**Issue by mattkerrison** [sysadminsmedia/homebox#33]
### What is the problem you are trying to solve with this feature?

This project would benefit lots of people that use Home assistant. Creating an addon for it isn't too involved, I've seen a number of people chasing a home inventory tool and settling for Grocy as it has an addon available.

### What is the solution you are proposing?

Create a HA addon of Homebox:
https://developers.home-assistant.io/docs/add-ons/

### Contributions

- [X] I have searched through existing issues and feature requests to see if my idea has already been proposed.
- [X] If this feature is accepted, I would be willing to help implement and maintain this feature.
- [ ] If this feature is accepted, I'm willing to sponsor the development of this feature.

**Comment by mattkerrison** [comment]
Update on this, I have an addon working with all environment variables configurable in the addon's configuration page, as well as the data directory which contains image data & the DB in the addon_configs directory.

The only thing I haven't been able to get working is ingress but that seems related to the multiple reverse proxy issues raised in the old repo. I've been able to work around this as I used the cloudflared addon to access home assistant externally and it can proxy for other services as well (with added benefit of providing SSO auth before forwarding requests. It'd be good to have the ability to use Auth headers so can bypass the login screen but that's another topic.

I'll create a PR in coming days, it's a little rough around the edges but being able to run it off the same box as Home Assistant and not have to rely on setting up proxmox to do so is great.

**Comment by HydrelioxGitHub** [comment]
I would love to use my home assistant instance to run home box as both of them are related to house management. 

Is this still a work in progress ?

**Comment by LelouBil** [comment]
Here's the reference about the headers available.
https://developers.home-assistant.io/docs/add-ons/security/#authenticating-a-user-when-using-ingress
Is it currently possible to auto login/auto create users with headers in homebox ?

**Comment by tankerkiller125** [comment]
As far as I'm aware we do not support header-based login, we do plan to have OIDC support (there's a draft PR for that), but nothing regarding header-based logins.

**Comment by LelouBil** [comment]
Would you be open to a PR for this ? Do you think it would be hard to implement for a new contributor?

**Comment by tankerkiller125** [comment]
We would be open to a PR for it, seems like something people would want/need. I'm not sure on the difficulty level for a new contributor, I don't think it would be too bad?

**Comment by KevinKTM** [comment]
Home assistant addon would be so excellent. Its something missing - and in the more generic sense, maintenance logs for vehicles, lawnmowers etc.

**Comment by tankerkiller125** [comment]
@mattkerrison Any luck overall and anything PR wise to get this rolled into Home Assistant? I haven't had time myself to dig into this between work and overall Homebox development.

**Comment by alvin96** [comment]
It would be very helpful if this come real. Looking forward to it!

**Comment by xeor** [comment]
Looks like the header-pr is coming along nicely :D

@mattkerrison, I didnt find your work on the addon itself. Is it still around.. Did you meet any blockers?

**Comment by Artiom97es** [comment]
oh nice, didnt even think about home assitant integration

**Comment by jscarrott** [comment]
I hacked together an addon for home assistant a while ago and just updated it as I've started using homebox. It lives [here](https://github.com/jscarrott/homebox-addon) and is obviously still very rough, but it does work. Not much more help  than following the docs, really, but if there is interest I might tidy it up.

**Comment by tankerkiller125** [comment]
@jscarrott just as a note, we'll be moving ARM build to *-arm tagged container releases. (You can see this with main and vnext). We've done this due to the insane build times (10-15x longer than x86) for the ARM builds. Which made it hard to validate that Docker builds for PRs were passing quickly, and also significantly delayed the speed at which we were able to get new releases to people.

**Comment by HawkanForce** [comment]
@jscarrott I tried to install the addon but I get this error "Failed to install add-on
Can't install ghcr.io/jscarrott/amd64-addon-example:2.0.2: 500 Server Error for http+docker://localhost/v1.47/images/create?tag=2.0.2&fromImage=ghcr.io%2Fjscarrott%2Famd64-addon-example&platform=linux%2Famd64: Internal Server Error ("Head "https://ghcr.io/v2/jscarrott/amd64-addon-example/manifests/2.0.2": unauthorized")"

**Comment by jscarrott** [comment]
Ah the amd64 docker registry was still set to private it should work now. I originally setup it up so long ago and only used the arm version I had forgotten about access control.

**Comment by jscarrott** [comment]
I might have some time in the next week or so to sort out some of the cruft left over from the original example template. I'm also happy to move the repo into an org if people want to base the addon on it.

**Comment by HawkanForce** [comment]
thank you @jscarrott now I got it installed. I'll play around with it a bit :)

**Comment by HawkanForce** [comment]
The addon seems to work perfectly. Helps a lot to be able to have the homebox on home assistant :) Thank you!

**Comment by tankerkiller125** [comment]
@jscarrott if this is something you're willing/wanting to maintain, I think we'd probably be interested in bringing it into our org with you as the maintainer. (For whatever maintenance Home Assistant extensions might need)

**Comment by jscarrott** [comment]
Yeah that sounds good to me, maintaining the add on should be pretty easy, just rebuild after each new version and update the changelog. There is probably a couple of hours work cleaning it up and adding the ingress so it can be accessed from the home assistant side panel.

**Comment by tankerkiller125** [comment]
Sorry for the delay on this @jscarrott, I've forked your repo into our org, and added you as a maintainer for that project.

**Comment by kenny-f** [comment]
Cannot access this addon over https. 
I installed this addon but was unable to access it via `https://homeassistant.local:7745`

I have verified this works on a fresh instance on HA without https enabled. 

I'm not sure where in the infra stack this need/could be solved, maybe adding [ingress](https://developers.home-assistant.io/docs/add-ons/presentation#ingress) like the adguard home addon?
